### DISCUSSION TITLE
Politique de sécurité Vaultwarden : enforce MFA cross-organisation

### DISCUSSION THREAD

**Discussion by Rhmaric** [betagouv/cap#27] [Besoin]
### Besoin

Bonjour,

J'ai été surpris de découvrir que l'ajout d'un MFA n'était pas obligatoire sur la connexion à VaultWarden.

Il s'agit d'une bonne pratique qu'il serait bénéfique de rendre obligatoire au niveau de la configuration du serveur (si cela est possible bien évidemment). Aujourd'hui, c'est à la discrétion du gestionnaire de l'organisation, sachant que des comptes individuels non rattaché à une orga existent.

Etant possible d'utiliser VaultWarden comme une application d'authentification (fournisseur de code OTP). On pourrait considérer qu'un compte VaultWarden sans MFA constituerait une faille de sécurité permettant de bypasser la logique MFA (un seul mot de passe permettant d'accéder aux 2 informations d'authentification : login et code OTP).

Sur ce dernier sujet, je me demande même si l'utilisation de VaultWarden comme application OTP est une bonne pratique...?

### Solutions envisageables

- Rendre obligatoire la configuration d'un MFA au niveau de la configuration du serveur (si cela est possible bien évidemment)
- Communiquer sur les bonnes pratiques à propos de l'utilisation de VaultWarden comme fournisseur de seconde d'authentification (code OTP).

### Informations complémentaires

- [ ] J'avais déjà entendu ce besoin au sein de la communauté beta.gouv
- [X] Je me porte volontaire pour participer à la mise en place de la future solution choisie

**Comment by revolunet** [comment]
Oui c'est un sujet.
Apparemment il n'est pas possible de forcer le MFA sur vaultwarden (shocking).

Du coup il est prévu de faire un mailing et suivre la mise en place par les utilisateurs (on peut savoir qui ne l'a pas encore mis en place)

**Reply by revolunet** [reply]
~~En fait le mailing va se faire sur les owners d'organisation qui peuvent eux forcer le MFA de leur orga.~~

Notre [doc VaultWarden](https://doc.incubateur.net/communaute/les-outils-de-la-communaute/autres-services/vaultwarden) est modifiable ici : https://github.com/betagouv/doc.incubateur.net-communaute/blob/master/les-outils-de-la-communaute/autres-services/vaultwarden/README.md

Envoyé aux 233 utilisateurs sans MFA ce jour

**Reply by Rhmaric** [reply]
Oui effectivement si ce n'est pas possible d'activer sur l'ensemble du server alors il y a aussi la solution de contacter les owners des orga pour qu'ils activent l'option et ensuite faire le suivi de l'activation.
Par contre cela ne règle pas vraiment le sujet sur les comptes individuels. Si l'information est connue, peut être qu'un rappel automatisé par mail pourrait être une solution (voir un message sur mattermost).

**Reply by revolunet** [reply]
WIP: https://github.com/orgs/betagouv/projects/82/views/14?pane=issue&itemId=86528519&issue=betagouv%7Canimation-communaute%7C520
